tutorial.md

```diff
@@ -0,0 +1,305 @@
+
+```
+order_id,customer_id,purchase_date,amount
+ORDER1,CUST1,1/1/2021,100
+ORDER2,CUST2,2/2/2022,200
+ORDER3,CUST3,3/3/2023,300
+```
+
+### Set up Snowflake
+
+(TODO: Add some generic instructions for setting up a trial Snowflake instance)
+
+## Install Astro on your local machine
+
+* [Install the Astro CLI](https://docs.astronomer.io/astro/install-cli), following the additional steps to [create an Astro project.](https://docs.astronomer.io/astro/create-project)
+
+* In the project directory you just created, create a `.env` file using your favorite file editor:
+
+```shell
+AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True
+AIRFLOW__ASTRO__SQL_SCHEMA=<snowflake_schema>
+```
+
+
+* Note: prior to running `astrocloud dev start` you'll need to navigate to your Docker Desktop application and double-click on it.
+
+* Check out your Astronomer Airflow UI at `http://localhost:8080/`
+
+## Setup Airflow connections
+
+In your Astronomer Airflow UI select Admin-->Connectors
+
+![connectors](http://localhost:8888/files/Screen%20Shot%202022-05-17%20at%204.05.00%20PM.png?_xsrf=2%7C7f8215bb%7C78c6f1df99528f5087b9d85f0eb5de75%7C1650662527)
+
+### Connect S3 to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+Set these fields:
+
+* Connection Id: `aws_default`
+* Connection Type: `S3`
+* Extra: `{"aws_access_key_id": "<your_access_key", "aws_secret_access_key": "<you_secret_access_key"}`
+
+### Connect Snowflake to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+* Connection Id: `snowflake_default`
+* Connection Type: `Snowflake`
+* Host: `https://<account>.<region>.snowflakecomputing.com/`. This is the URL where you can log into your Snowflake account
+* Schema:
+* Login:
+* Password:
+* Account:
+* Database:
+* Region: (something like us-east-1)
+* Role:
+* Warehouse:
+
+
+***
+# Create and populate some tables in Snowflake
+
+We'll create some auxiliary tables in Snowflake and populate with a small amount of data for our ETL example.
+
+* Create and populate a customer table to join with an orders table that we create with the Astro Python SDK:
+
+```sql
+
+CREATE OR REPLACE TABLE customers_table (customer_id CHAR(10), customer_name VARCHAR(100), type VARCHAR(10) )
+
+INSERT INTO customers_table (CUSTOMER_ID, CUSTOMER_NAME,TYPE) VALUES ('CUST1','NAME1','TYPE1'),('CUST2','NAME2','TYPE1'),('CUST3','NAME3','TYPE2')
+```
+
+* Create and populate a reporting table into which we'll merge our transformed data:
+
+```sql
+
+CREATE OR REPLACE TABLE reporting_table (
+    CUSTOMER_ID CHAR(10), CUSTOMER_NAME VARCHAR(100), ORDER_ID CHAR(10), PURCHASE_DATE DATE, AMOUNT FLOAT, TYPE CHAR(10));
+    
+INSERT INTO reporting_table (CUSTOMER_ID, CUSTOMER_NAME, ORDER_ID, PURCHASE_DATE, AMOUNT, TYPE) VALUES 
+('INCORRECT_CUSTOMER_ID','INCORRECT_CUSTOMER_NAME','ORDER2','2/2/2022',200,'TYPE1'),
+('CUST3','NAME3','ORDER3','3/3/2023',300,'TYPE2'),
+('CUST4','NAME4','ORDER4','4/4/2022',400,'TYPE2')
+```
```

Comment:
Could we reduce the user overhead by using Astro's load task and not having this step?

Reply:
I was trying to make the example code as short as possible, so there's a tradeoff with that goal vs. doing stuff outside in the code, in Snowflake itself.

Since we already have one `load_file` example in the code, any additional ones are somewhat superfluous for the purposes of demonstrating `load_file`

But I could be convinced either way on this issue.